scipy/optimize/_trustregion_exact.py

```diff
@@ -208,10 +208,18 @@ class IterativeSubproblem(BaseQuadraticSubproblem):
     # As recommended there it value is fixed in 0.01.
     UPDATE_COEFF = 0.01
 
+    # The subproblem may iterate infinitely for problematic
+    # cases (see https://github.com/scipy/scipy/issues/12513).
+    # When the `maxiter` setting is None, we need to apply a
+    # default. An ad-hoc number (though tested quite extensively)
+    # is 25, which is set below. To restore the old behavior (which
+    # potentially hangs), this parameter may be changed to zero:
+    MAXITER_DEFAULT = 25  # zero means infinite.
```

Comment:
this is the behaviour when `subproblem_maxiter=None`, correct? If so, could you document that where you have added `subproblem_maxiter` to the docstring? My naïve interpretation of "... max iterations equals None" is _no maximum_ i.e. `MAXITER_DEFAULT = 0`.